src/types/interface.ts

```diff
@@ -1,12 +1,11 @@
 export interface IProfileData {
   uid: string;
-  username: string;
-  firstName: string;
-  lastName: string;
+  displayName: string;
+  fullName: string;
   email: string;
   address: string;
-  phone: string;
-  photo: string;
+  phoneNumber: string;
+  photoURL: string;
 }
 
 export interface IUserSignInData {
```

Comment:
```js
export interface IUpdateUser {
  name: string;
  photoURL: string;
  phoneNumber: string;
}
```
This piece of code can be reused like this 

```suggestion

export interface IUpdateUser
  extends Pick<IProfileData, "photoURL" | "phoneNumber"> {
  name: string;
}

```